src/selectors/layers.test.js

```diff
@@ -34,8 +34,8 @@ describe('Selectors', () => {
       }, {});
 
       nodes.forEach(node => {
-        // we don't need to check y/height positions if the layer isn't there.
-        if (node.layer === null) {
+        // Wwe don't need to check y/height positions if the layer isn't there or if the 'layer' field does not exists ( this is the case for 'task' type nodes ).
+        if (!node.layer) {
```

Comment:
This comment change seems to just repeat what it already said? Is that needed?

Also the test `if (!node.layer)` still fails for the case of `node.layer = 0`.

For that reason I'd prefer `if (node.layer === null || typeof node.layer === 'undefined')` in cases like this.

Reply:
thanks for the comments!  I will modify the comment change.

In terms of the check, currently on just using `(node.layer === null || typeof node.layer === 'undefined')` as a test would cause this test to fail given that `node.layer` field does not exist for some nodes - it is not simply testing for the case where `node.layer` as either assigned as `null` or `undefined` as previously set up in the animals dataset. And as we mentioned above, since `!node.layer` covers all negative case, hence we would not need to add in any extra test to specifically test for the `null` and `undefined` cases.

While I understand your concern for the `node.layer = 0` case, since this check is only within a test case where we are testing against a specfici dataset with no cases of `node.layer = 0`, hence there isn't any particular need to cater for this case.

Reply:
Thanks @bru5 - I've changed the comment and modified the check. ( the current check is only used within a test against a specific dataset without any cases of `node.layer = 0` hence there isn't a particular need to cater for this case, but nonetheless I have changed the check to your suggested format just in case.)